x-pack/plugins/maps/public/actions/layer_actions.ts

```diff
@@ -280,7 +282,12 @@ export function updateSourceProp(
   value: unknown,
   newLayerType?: LAYER_TYPE
 ) {
-  return async (dispatch: ThunkDispatch<MapStoreState, void, AnyAction>) => {
+  return async (
+    dispatch: ThunkDispatch<MapStoreState, void, AnyAction>,
+    getState: () => MapStoreState
+  ) => {
+    const layer = getLayerById(layerId, getState());
+    const previousFields = await (layer as IVectorLayer).getFields();
```

Comment:
You are going to need a guard since this method will be called non-vector layers and getFields will not exist for those layers.

Reply:
lmk what you think. I'd split up in two calls, rather than having code branch-split in the middle of a block.

Reply:
lmk what you think. I've split up in two calls, rather than having code branch-split in the middle of a block

Reply:
I still like the branch splitting route to avoid the duplication of updating source property. Now, if we update how to modify a source property, you have to ensure both places are properly updated instead of just a single location.

But this is more of a nit and you can disregard as well.